sdk/ml/azure-ai-ml/azure/ai/ml/operations/_capability_hosts_operations.py

```diff
@@ -0,0 +1,345 @@
+        ops_logger.update_info(kwargs)
+        self._all_operations = all_operations
+        self._capability_hosts_operations = service_client_10_2024.capability_hosts
+        self._workspace_operations = service_client_10_2024.workspaces
+        self._credentials = credentials
+        self._init_kwargs = kwargs
+
+    @experimental
+    @monitor_with_activity(ops_logger, "CapabilityHost.List", ActivityType.PUBLICAPI)
+    @distributed_trace
+    def list(self, **kwargs: Any) -> Iterable[CapabilityHost]:
+        """List capability hosts in a Hub or Project workspace.
+
+        :return: List of CapabilityHost objects.
+        :rtype: ~typing.Iterable[~azure.ai.ml.entities._workspace._ai_workspaces.capability_host.CapabilityHost]
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/ml_samples_capability_host.py
+                :start-after: [START capability_host_list_operation]
+                :end-before: [END capability_host_list_operation]
+                :language: python
+                :dedent: 8
+                :caption: List example.
+        """
+
+        return self._capability_hosts_operations.list(
+            resource_group_name=self._resource_group_name,
+            workspace_name=self._workspace_name,
+            **kwargs,
+        )
+
+    @experimental
+    @monitor_with_activity(ops_logger, "CapabilityHost.Get", ActivityType.PUBLICAPI)
+    @distributed_trace
+    def get(self, name: str, **kwargs: Any) -> CapabilityHost:
+        """Retrieve a capability host resource.
+
+        :param name: The name of the capability host to retrieve.
+        :type name: str
+        :raises ~azure.ai.ml.exceptions.ValidationException: Raised if project name or hub name
+            not provided while creation of MLClient object in workspacename param.
+            Details will be provided in the error message.
+        :raises ~azure.ai.ml.exceptions.ValidationException: Raised if Capabilityhost name is not provided.
+            Details will be provided in the error message.
+        :return: CapabilityHost object.
+        :rtype: ~azure.ai.ml.entities._workspace._ai_workspaces.capability_host.CapabilityHost
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/ml_samples_capability_host.py
+                :start-after: [START capability_host_get_operation]
+                :end-before: [END capability_host_get_operation]
+                :language: python
+                :dedent: 8
+                :caption: Get example.
+        """
+
+        self.__validate_workspace_name()
+
+        rest_obj = self._capability_hosts_operations.get(
+            resource_group_name=self._resource_group_name,
+            workspace_name=self._workspace_name,
+            name=name,
+            **kwargs,
+        )
+
+        capability_host = CapabilityHost._from_rest_object(rest_obj)
+
+        return capability_host
```

Comment:
```suggestion
        self.__validate_workspace_name()

        return self._capability_hosts_operations.get(
            resource_group_name=self._resource_group_name,
            workspace_name=self._workspace_name,
            name=name,
            cls=CapabilityHost._from_rest_object,
            **kwargs,
        )

```

Reply:
we cant use cls for get operation.. get operation returns the restclient object instead of our user defined object.

Reply:
that's why we have the lambda to do the deserialization. The behavior is the same for all CRUD operations. If it doesn't work, let's figure out why.

Reply:
lambda performs the serialization but we need to return entities.capabilityhost where as restclient.get returns the restclient.capabilityhost hence, its conversion error here. 

get returns _models.CapabilityHost (restclient.capabilityhost) and not the entities.capabilityhost

Reply:
as per discussion over the teams call, this change is required.